datajoint/expression.py

```diff
@@ -119,17 +121,37 @@ def where_clause(self):
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
         )
 
+    def sorting_clauses(self):
+        if not self._top:
+            return ""
+        limit = self._top.limit
+        order_by = self._top.order_by
+        offset = self._top.offset
```

Comment:
Do we need to create these variables?

Reply:
Probably not anymore, since they aren't occurring as often in this context after moving the type handling to the `__post_init__`.